clients/python-wrapper/lakefs/branch.py

```diff
@@ -219,14 +219,14 @@ def delete(self) -> None:
         with api_exception_handler():
             return self._client.sdk_client.branches_api.delete_branch(self._repo_id, self._id)
 
-    def revert(self, reference_id: str, parent_number: Optional[int] = None) -> None:
+    def revert(self, reference: ReferenceType, parent_number: Optional[int] = None) -> Commit:
```

Comment:
So since `str` is a subtype of `ReferenceType`, we are compatible there (any previously legal input is still legal now, plus `Reference` and `Commit` objects). 
Since we are now returning a value that was previously None (and thus not interesting), we should be good there, too. Only problem I see is with people passing `reference_id` as a keyword argument to `Branch.revert()`.

Do you think that's common? I would love to at least keep that increased flexibility in typing.

Reply:
You are of course correct, on all counts.  Let's fix everything: have an argument `reference` that is a string and alias `reference_id` to that argument.  There are [decorators that can do this](https://stackoverflow.com/a/29374525/192263), but probably just 

```suggestion
    def revert(self, _reference: ReferenceType, parent_number: Optional[int] = None, reference: Optional[ReferenceType] = None, reference_id: Optional[ReferenceType] = None) -> Commit:
      # Allow reference_id, the deprecated old argument, as a keyword arg.
      reference = _reference or reference or reference_id
```

will mostly do the trick.

@N-o-Z can we go there?  Because realistically this _is_ a mistyping, it makes a lot of sense to allow the second arg to be a reference and not just an ID, and Python handling of function args is spectacularly dangerous for software engineering.  Otherwise we break backwards compatibility in this small way, or keep a confusing arg name.

Reply:
Sure, that's a beautiful solution!  Your version will be entirely backwards compatible :-)

```suggestion
    def revert(self, reference: ReferenceType, parent_number: Optional[int] = None, *, reference_id: Optional[str] = None) -> Commit:
      # Handle reference_id as a deprecated alias to reference.
      reference = reference or reference_id
```

Reply:
Thanks, I took this over in [c253b34](https://github.com/treeverse/lakeFS/pull/7353/commits/c253b347777c481b70ce93ada3c88d7653e5d9b3).